code/modules/vehicles/aircraft.dm

```diff
@@ -0,0 +1,1297 @@
+		STOP_PROCESSING(SSobj, src)
+		addtimer(CALLBACK(src, .proc/shutdown_procedures), 1 SECONDS)
+	else
+		balloon_alert(pilot, "Out of fuel!")
+
+///Series of actions taken when the aircraft is no longer flying
+/obj/vehicle/sealed/aircraft/proc/shutdown_procedures()
+	set_flying(FALSE)
+	coverage = initial(coverage)
+	layer = initial(layer)
+	remove_filter("flight_shadow")
+	icon_state = initial(icon_state)
+	for(var/mob/M in occupants)
+		M.set_flying(FALSE)
+	if(pilot)
+		pilot.update_action_buttons()	//For updating the toggle engine button
+	var/turf/landing_zone = get_turf(src)
+	if(landing_zone.density)
+		var/no_valid_turf = TRUE
+		for(var/turf/target in get_adjacent_open_turfs(src))
+			if(!is_blocked_turf(target))	//Try to find a valid open turf
+				throw_at(target, 2, 3, spin = TRUE)	//Throw the vehicle at this open turf
+				take_damage(FLYING_CRASH_DAMAGE)
+				no_valid_turf = FALSE
+				break
+		if(no_valid_turf)
+			obj_destruction()	//If no valid open turfs nearby, aircraft goes up in flames
+	if(LAZYLEN(landing_zone.contents))	//Let's crush anything this lands on!
+		for(var/atom/A in landing_zone.contents)
+			if(A == src)	//Don't splat ourselves!
+				continue
+			if(ismob(A))
+				var/mob/living/victim = A
+				victim.apply_damage(40)
+				victim.Stun(2 SECONDS)
+				visible_message(span_alert("[victim.name] was crushed under the weight of [src]!"))	//Important to keep as a message for logging
+				playsound(src, 'sound/effects/clownstep2.ogg', 100, FALSE, 7)	//Funny, but placeholder until I find a proper squish/splat sound
+			else if(isobj(A))
+				var/obj/victim = A
+				if(!CHECK_BITFIELD(victim.resistance_flags, INDESTRUCTIBLE))
+					victim.obj_destruction()
+					visible_message(span_alert("[victim.name] was crushed under the weight of [src]!"))	//Important to keep as a message for logging
+
+//Aircraft burn fuel at all times while flying, even if only hovering
+/obj/vehicle/sealed/aircraft/process()
+	current_fuel--
+	update_icon()
+	if(current_fuel <= 0)
+		playsound(src, 'sound/mecha/lowpower.ogg', 100, FALSE, 7)	//Should play only to the pilot but we have no aircraft descending sound yet
+		toggle_engine()
+	if(pilot.stat)	//The dead can't fly!
+		balloon_alert_to_viewers("Pilot control lost!")	//To let people know the pilot isn't awake
+		toggle_engine()
+	if(COOLDOWN_CHECK(src, enginesound_cooldown))
+		COOLDOWN_START(src, enginesound_cooldown, 3.5 SECONDS)
+		playsound(src, 'sound/effects/tadpolehovering.ogg', 60, TRUE, 10)
+	if(being_destroyed)
+		setDir(pick(LeftAndRightOfDir(dir, TRUE)))
+
+/obj/vehicle/sealed/aircraft/Moved(atom/old_loc, movement_dir, forced, list/old_locs)
+	. = ..()
+	if(!LAZYLEN(occupants))
+		return
+	if(current_fuel > 0)
+		current_fuel--
+
+/obj/vehicle/sealed/aircraft/relaymove(mob/living/user, direction)
+	if(!CHECK_BITFIELD(flags_pass, FLYING))
+		if(!TIMER_COOLDOWN_CHECK(src, "aircraft anti spam"))
```

Comment:
missing defines, why is this a timer CD check

Reply:
I just copied how mech code did non-spammy messages.